src/components/StationPicker/StationPicker.tsx

```diff
@@ -79,27 +87,25 @@ const StationPicker = (props: Props) => {
                 window.removeEventListener("click", closeOnClickOutsideHandler);
             };
         }
-    }, [disclosure.visible]);
```

Comment:
`store.getState()` won’t trigger a component re-render, which may lead to bugs. The official equivalent to `disclosure.visible` is:

```jsx
import { useStoreState } from "@ariakit/react";

const disclosure = useDisclosureStore();
// visible is equivalent to Reakit's disclosure.visible
const visible = useStoreState(disclosure, "open");
```

See [Watching a specific state property](https://ariakit.org/guide/component-stores#watching-a-specific-state-property) for more info.